src/components/Quotes.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+
+const Quotes = () => {
+  const [quote, setQuote] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setLoading(true);
+      try {
+        const getData = await fetch('https://api.api-ninjas.com/v1/quotes', {
+          method: 'GET',
+          headers: { 'X-Api-Key': 'XHrTP6DOL5nJ3iiJTwrfMg==OpEmLQCgPTqaBmcs' },
+          contentType: 'application/json',
+        });
+        const response = await getData.json();
+        setQuote(response);
+      } catch {
+        setHasError(true);
+      }
+      setLoading(false);
+    };
+    fetchData();
+  }, [setQuote, setLoading]);
```

Comment:
- [ ] Good job creating the Quotes component :clap: , I think it will be better to use an empty array in the dependency as it will make sure the effect runs only once when the component is initially rendered or mounted on the screen, currently, the setQuote and setLoading functions are included in the dependency array, which is incorrect because they are not used inside the fetchData function. Adding these functions as dependencies will cause the useEffect to run every time these functions change, which will cause unnecessary re-renders and possibly even an infinite loop. Therefore, it is recommended to use an empty dependency array in this case since fetchData doesn't rely on any other variables or functions. please fix this by adding an empty array `[ ]` in the dependency

Reply:
Could you please explain what do you mean by "which is incorrect because they are not used inside the fetchData function"?

Reply:
Hi @sameedmohsin28, Kindly reach out to the  reviewer for more clarification 👍🏿

Reply:
hi @sameedmohsin28, In this case, including setQuote and setLoading in the dependency array is not incorrect, but it is not necessary. It is because the functions setQuote and setLoading will always have the same reference throughout the component's lifecycle, so there is no need to include them in the dependency array. they are also functions responsible for changing the state of your components i.e. loading, quotes and error.